tfx/examples/cifar10/cifar10_utils_native_keras.py

```diff
@@ -0,0 +1,338 @@
+  eval_dataset = _input_fn(fn_args.eval_files, tf_transform_output, False,
+                           _EVAL_BATCH_SIZE)
+
+  mirrored_strategy = tf.distribute.MirroredStrategy()
+  with mirrored_strategy.scope():
+    model, base_model = _build_keras_model()
+
+  # Our training regime has two phases: we first freeze the backbone and train
+  # the newly added classifier only, then unfreeze part of the backbone and
+  # fine-tune with classifier jointly.
+  steps_per_epoch = _TRAIN_DATA_SIZE / _TRAIN_BATCH_SIZE
+  total_epochs = int(fn_args.train_steps / steps_per_epoch)
+  classifier_epochs = int(total_epochs / 2)
+  finetune_epochs = total_epochs - classifier_epochs
+
+  absl.logging.info('Start training the top classifier')
+  # Freeze the whole MobileNet backbone and train the top classifer only
+  _freeze_model_by_percentage(base_model, 1.0)
+  # We need to recompile the model because layer properties have changed
+  model.compile(
+      loss='sparse_categorical_crossentropy',
+      optimizer=tf.keras.optimizers.RMSprop(lr=3e-4),
+      metrics=['sparse_categorical_accuracy'])
+  model.summary(print_fn=absl.logging.info)
+
+  model.fit(
+      train_dataset,
+      epochs=classifier_epochs,
+      steps_per_epoch=steps_per_epoch,
+      validation_data=eval_dataset,
+      validation_steps=fn_args.eval_steps
+  )
+
+  absl.logging.info('Start fine-tuning the model')
+  # Unfreeze the top MobileNet layers and do joint fine-tuning
+  _freeze_model_by_percentage(base_model, 0.9)
+  # We need to recompile the model because layer properties have changed
+  model.compile(
+      loss='sparse_categorical_crossentropy',
+      optimizer=tf.keras.optimizers.RMSprop(lr=5e-5),
+      metrics=['sparse_categorical_accuracy'])
+  model.summary(print_fn=absl.logging.info)
+
+  model.fit(
+      train_dataset,
+      epochs=finetune_epochs,
+      steps_per_epoch=steps_per_epoch,
+      validation_data=eval_dataset,
+      validation_steps=fn_args.eval_steps
+  )
+
+  # Save the trained model in TFLite format for serving
+  signatures = {
+      'serving_default':
+          _get_serve_image_fn(
+              model).get_concrete_function(
+                  tf.TensorSpec(
+                      shape=[None, 224, 224, 3],
+                      dtype=tf.float32,
+                      name=transformed_name(IMAGE_KEY)
+                      ))
+  }
+
+  temp_saving_model_dir = os.path.join(fn_args.serving_model_dir, 'temp')
+  model.save(temp_saving_model_dir, save_format='tf', signatures=signatures)
+
+  tfrw = rewriter_factory.create_rewriter(
+      rewriter_factory.TFLITE_REWRITER, name='tflite_rewriter',
+      enable_experimental_new_converter=True)
+  converters.rewrite_saved_model(temp_saving_model_dir,
+                                 fn_args.serving_model_dir,
+                                 tfrw,
+                                 rewriter.ModelType.TFLITE_MODEL)
+  tflite_model_path = os.path.join(fn_args.serving_model_dir, 'tflite')
```

Comment:
This is great. A couple of things (i) for 'tflite', can we make this a top-level constant again. Also. Lets leave a todo here to extend the tflite_rewriter to also add the writing of metadata.